wp-includes/template.php

```diff
@@ -171,3 +171,47 @@ function wp_service_worker_error_details_template( $output = '' ) {
 function wp_service_worker_error_message_placeholder() {
 	echo '<p><!--WP_SERVICE_WORKER_ERROR_MESSAGE--></p>';
 }
+
+/**
+ * Reload the offline page and check if user comes online.
+ *
+ * @since 0.7
+ */
+function wp_service_worker_offline_page_reload() {
+	if ( ! is_offline() ) {
```

Comment:
It would be useful to also do the reload in the case of server being down.

Also, something else that comes to mind: what happens if the error page is served in response to a POST request? If so, each reload will cause a re-POST message from the browser and that's probably not desired.

```suggestion
	if ( ! is_offline() || ! is_500() || 'GET' !== $_SERVER['REQUEST_METHOD'] ) {
```

Reply:
Thanks for pointing this out. For sure we need to check the method of the request. I have implemented these changes.

Reply:
As part of #728, I just realized that this check will never do anything:

https://github.com/GoogleChromeLabs/pwa-wp/blob/124baabe30f70b718960fa18158f10a54943287e/wp-includes/template.php#L181-L184

This is because the offline template is always fetched via a `GET` request by the service worker when it is installing, so it will never not be `GET`. We need an alternative way of preventing the code from running on responses to non-GET requests.

Contrary to what I thought above, each reload does _not_ cause the browser re-POST message: https://github.com/GoogleChromeLabs/pwa-wp/pull/728#issuecomment-1099718487

Reply:
I believe I've addressed this via https://github.com/GoogleChromeLabs/pwa-wp/pull/763

Reply:
See https://github.com/GoogleChromeLabs/pwa-wp/pull/763